src/message_sender.c

```diff
@@ -562,7 +562,6 @@ static void on_link_state_changed(void* context, LINK_STATE new_link_state, LINK
         {
             /* User initiated transition, we should be good */
             set_message_sender_state(message_sender_instance, MESSAGE_SENDER_STATE_IDLE);
-            indicate_all_messages_as_error(message_sender_instance);
```

Comment:
>indicate_all_messages_as_error [](start = 12, length = 30)

Thanks for looking into this.
I looked at the change and at what we have in link.c and I think we should go one step further to cover as much as we can of the cases.

I think 
remove_all_pending_deliveries(link_instance, true);
should be called whenever the link state is set to DETACHED/ERROR.

That would be link.c ln 280, ln 330, ln. 549, ln. 555.

Or directly in the set_link_state if the link_state is DETACHED or ERROR.

In that case the indicate_all_messages_as_error call actually can stay, as it won't hurt, but probably will be hit with an empty list of messages.

I would have made the change by myself, but I wanted to ask you if you can check that it also solves your case.